Src/CSharpier/SyntaxPrinter/CSharpierIgnore.cs

```diff
@@ -26,29 +20,42 @@ FormattingContext context
     ) where T : SyntaxNode
     {
         var statements = new List<Doc>();
+        var unFormattedCode = new StringBuilder();
         var printUnformatted = false;
 
         foreach (var node in list)
         {
             if (Token.HasLeadingComment(node, "// csharpier-ignore-end"))
             {
+                statements.Add(unFormattedCode.ToString().Trim());
+                unFormattedCode.Clear();
                 printUnformatted = false;
             }
             else if (Token.HasLeadingComment(node, "// csharpier-ignore-start"))
             {
                 printUnformatted = true;
             }
 
-            statements.Add(
-                printUnformatted ? PrintWithoutFormatting(node) : Node.Print(node, context)
-            );
+            if (printUnformatted)
+            {
+                unFormattedCode.Append(PrintWithoutFormatting(node));
+            }
+            else
+            {
+                statements.Add(Node.Print(node, context));
+            }
+        }
+
+        if (unFormattedCode.Length > 0)
+        {
+            statements.Add(unFormattedCode.ToString().Trim());
         }
 
         return statements;
     }
 
-    public static Doc PrintWithoutFormatting(SyntaxNode syntaxNode)
+    public static string PrintWithoutFormatting(SyntaxNode syntaxNode)
     {
-        return syntaxNode.GetText().ToString().Trim();
+        return syntaxNode.GetText().ToString();
```

Comment:
So, I think this means we won't even trim trailing whitespace for `// csharpier-ignore` blocks.
Is this intentional?

Reply:
I think it used to unintentionally trim trailing whitespace, which also caused the problems with removing lines when it shouldn't. Or maybe I did intentionally make it do that? I didn't have a test for it so I'm not sure.
Either way now it trims trailing properly(with tests!), and fixes end of lines. But I don't see a good way to test the end of line part. The test file could have CRLF or LF depending on how the repo gets cloned.